ILRepack/RepackImporter.cs

```diff
@@ -582,6 +582,95 @@ private void CloneTo(MethodBody body, MethodDefinition parent)
 
                 nb.ExceptionHandlers.Add(neh);
             }
+
+            CopyDebugInformation(body.Method, parent);
+        }
+
+        public void CopyDebugInformation(MethodDefinition input, MethodDefinition output)
+        {
+            CopySequencePoints(input, output);
+
+            // TODO: copy CustomDebugInformations
+            //if (input.DebugInformation.HasCustomDebugInformations)
+            //    CopyCustomDebugInformations(input, output);
+
+            output.DebugInformation.Scope = CopyDebugInformationScope(input.DebugInformation.Scope, output);
+        }
+
+        static void CopySequencePoints(MethodDefinition input, MethodDefinition output)
+        {
+            if (!input.DebugInformation.HasSequencePoints)
+                return;
+
+            foreach (var instr in input.Body.Instructions)
+            {
+                SequencePoint point = input.DebugInformation.GetSequencePoint(instr);
+                if (point == null)
+                    continue;
+
+                Instruction ni = output.Body.Instructions.First(i => i.Offset == instr.Offset);
+                SequencePoint nsp = new SequencePoint(ni, point.Document)
+                {
+                    StartLine = point.StartLine,
+                    EndLine = point.EndLine,
+                    StartColumn = point.StartColumn,
+                    EndColumn = point.EndColumn
+                };
+
+                output.DebugInformation.SequencePoints.Add(nsp);
+            }
+        }
+
+        static ScopeDebugInformation CopyDebugInformationScope(ScopeDebugInformation input, MethodDefinition parent)
+        {
+            if (input == null)
+                return null;
+
+            ScopeDebugInformation nsdi = new ScopeDebugInformation(
+                parent.Body.Instructions.First(i => i.Offset == input.Start.Offset),
+                input.End.IsEndOfMethod ? null : parent.Body.Instructions.First(i => i.Offset == input.End.Offset));
+
+            foreach (ScopeDebugInformation nested in input.Scopes)
+                nsdi.Scopes.Add(CopyDebugInformationScope(nested, parent));
+
+            if (input.Import != null)
+            {
+                var nidi = new ImportDebugInformation();
+                foreach (ImportTarget target in input.Import.Targets)
+                {
+                    var nt = new ImportTarget(target.Kind)
+                    {
+                        Namespace = target.Namespace,
+                        Alias = target.Alias,
+                        Type = target.Type == null ? null : parent.Module.ImportReference(target.Type),
+                        AssemblyReference = target.AssemblyReference == null ? null : parent.Module.Assembly.Name // TODO: determine where the type is - if it has not moved
```

Comment:
I am not sure about this code. I am copying the target assembly reference, but it most likely is not the same value. If this type is now merged into a single assembly, then the assembly is correct. But if it is a partial merge, then it will still point to the main assembly - even though the real assembly is not merged.

I think. I need to investigate this some more.